portfolio/src/main/webapp/SlideController.js

```diff
@@ -0,0 +1,23 @@
+import SlideHandler from '/SlideHandler.js';
+
+/** The controller handling all slideshows. */
+export default class SlideController {
+
+  constructor(document) {
+    /** 
+     * @private @const @type {SlideHandler} The service 
```

Comment:
Make types non-nullable by adding ! or likely to throw an NPE in the flip function if instantiation fails.

https://google.github.io/styleguide/jsguide.html#jsdoc-nullability